### ISSUE TITLE
Bug in total affected items when paginating groups in API

### ISSUE THREAD

**Issue by Selutario** [wazuh/wazuh#21912]
|Wazuh version|Component|
|---|---|
| 4.8.1 | Wazuh API |

## Description

The frontend team reported a problem with the offset value in 4.8.1. This is the issue:
- https://github.com/wazuh/wazuh-dashboard-plugins/issues/6388

Pagination is working fine up to group nº 500. However, when offset is used, the number of total affected items is wrong. For instance, if there are 504 groups:
- **offset 10, limit 10:** total_affected_items is `504`
- **offset 500, limit 10:** total_affected_items is `4`

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by GGP1** [comment]
## Update

Updated the way in which the `total_affected_items` number is calculated and performed various tests to verify that everything works as expected.

**Comment by Selutario** [comment]
After discussing it with other teams, we have decided to reopen this issue to apply a different solution:

Instead of pagination over the database, like shown below, we should get all the items (`limit=None`) and do pagination in the `process_array` again:
https://github.com/wazuh/wazuh/blob/9e96aed4826c570db43ea0860e8d6fbb8dfe6cbf/framework/wazuh/agent.py#L610-L611

This fix was initially discarded because the group hash calculation is done on demand, and doing it preemptively for all groups can lead to poor performance. However, for this performance loss to be significant, the environment should have more than 10k groups, which is far from real use cases. 

On the other hand, by obtaining all the items from the DB, we will solve the problem that still exists with the search and other filters on this endpoint.

**In summary:** we should leave the function as it was in [4.8.0](https://github.com/wazuh/wazuh/blob/34f20c6d801206080ccdc1dbf2eb2e225514f35b/framework/wazuh/agent.py#L596-L635), but obtain all the items from the DB.

**Comment by GGP1** [comment]
## Update

Applied the modifications suggested and tested the endpoint after the changes.
